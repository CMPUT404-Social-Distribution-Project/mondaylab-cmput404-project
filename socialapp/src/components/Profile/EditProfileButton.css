.edit-profile-modal .modal-header {
    border-bottom: 0 none;
    color: var(--teal);
}

.edit-profile-modal .btn-close {
    background-color: var(--teal);
    margin-right: 0rem;
}

.edit-profile-modal .modal-footer {
    border-top: 0 none;
}

.edit-profile-modal .form-control {
    background-color: var(--slightly-darker-blue);
}

.edit-profile-modal {
    min-width: 35em !important;
    width: 35em !important;
    max-width: 35em !important;
    transform: translateX(12%) !important;
    background-color: var(--dark-blue);
}

.edit-profile-modal .modal-body {
    padding: 0rem 1rem;
}

.edit-profile-button {
    margin-top: 2em;
    color: var(--dark-blue);
    background-color: var(--teal);
    border-radius: 2em;
    font-size: 14px;
    height: 2.5rem;
    width: 100%;
}

.edit-profile-button:hover {
    background-color: var(--teal-alt);
    color: var(--dark-blue);
}