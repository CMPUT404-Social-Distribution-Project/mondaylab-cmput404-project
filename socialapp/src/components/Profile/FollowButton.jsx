import React, { useEffect, useState, useContext } from "react";
import Button from "react-bootstrap/Button";
import AuthContext from "../../context/AuthContext";
import useAxios from "../../utils/useAxios";
import { useParams, useLocation } from "react-router-dom";
import { authorHostIsOurs, emptyNode } from "../../utils/utils";

// function authorInArray(id, array) {
//   // checks if the given author id is in the array
//   var i;
//   for (i = 0; i < array.length; i++) {
//     if (array[i].id === id) {
//       return true;
//     }
//   }
//   return false;
// }

export default function FollowButton(props) {
  // if not the current user, make the follow button visible
  const [isNotCurrentUser, setIsNotCurrentUser] = useState(true);
  const { baseURL } = useContext(AuthContext); // our api url http://127.0.0.1/service
  const api = useAxios(); // use this to add authorization header
  const currentAuthor = JSON.parse(localStorage.getItem("authTokens")).user; // the currently logged in author as an object
  const { author_id } = useParams(); // gets the author id in the url
  // Check if the currently logged in user is following the author they're viewing
  const [isFollowing, setIsFollowing] = useState(false);
  // set the follow state
  const [followState, setFollowState] = useState("notFollowing");

  useEffect(() => {
    if (currentAuthor.uuid === author_id) {
      // if the current user (user_id) has the same id in the url, don't show follow button
      setIsNotCurrentUser(false);
    }
  }, [isNotCurrentUser, useLocation().state]);

  useEffect(() => {
    const following = async (ApiURL, node) => {
      await api
        .get(`${ApiURL}authors/${author_id}/followers/${currentAuthor.uuid}`,
         { header: node.headers }
        )
        .then((response) => {
          if (response.data) {
            setIsFollowing(response.data);
            setFollowState("following");
          } else {
            setFollowState("notFollowing");
          }
        })
        .catch((error) =>{
          console.log(
            "Failed to get check if current author is following ", error.response
          );
        })
    };
    if (!authorHostIsOurs(props.authorViewing.host) && props.authorBaseApiURL !== null) {
      following(props.authorBaseApiURL, props.authorNode);
    } else {
      following(baseURL+'/', emptyNode);
    }
  }, [isFollowing, useLocation().state]);

  const handleClick = () => {
    if (followState === "notFollowing") {
      setFollowState("followSent");
      // Send a friend request object to the inbox of the author we're viewing
      api
        .post(`${props.authorBaseApiURL}authors/${author_id}/inbox/`, {
          type: "follow",
          summary: `${currentAuthor.displayName} wants to follow ${props.authorViewing.displayName}`,
          actor: currentAuthor,
          object: props.authorViewing,
        },
        {headers: props.authorNode.headers}
        )
        .then((response) => {
          console.log("Success sending a friend request: " + response);
        })
        .catch((error) => {
          setFollowState("notFollowing");
      });
    } else if (followState === "following") {
      // if we're already following, clicking will unfollow
      api
        .delete(
          `${props.authorBaseApiURL}authors/${author_id}/followers/${currentAuthor.uuid}`,
          {headers: props.authorNode.headers}
        )
        .then((response) => {
          console.log(
            "Success removing current author from viewing author's followers: " +
              response
          );
          setFollowState("notFollowing");
        })
        .catch((error) => {
          console.log("Failed to unfollow: " + error);
      });
    }
  };

  if (currentAuthor.uuid === props.authorViewing.uuid) {
    // don't show if viewing oneself's profile
    return <div className="empty-space" style={{ width: "3em" }} />;
  } else {
    return (
      <Button
        id="followButton"
        variant="primary"
        disabled={followState === "followSent"}
        onClick={followState !== "followSent" ? handleClick : null}
        style={{
          backgroundColor:
            followState === "following" ? "var(--orange)" : "var(--teal)",
        }}
      >
        {followState === "following" ? "Following" : ""}
        {followState === "followSent" ? "Sent" : ""}
        {followState === "notFollowing" ? "Follow" : ""}
      </Button>
    );
  }
}
