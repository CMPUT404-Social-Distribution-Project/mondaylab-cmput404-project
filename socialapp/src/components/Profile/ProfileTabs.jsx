import React from "react";
import Tab from "react-bootstrap/Tab";
import Tabs from "react-bootstrap/Tabs";
import { useNavigate } from "react-router-dom";

export default function ProfileTabs(props) {
  const navigate = useNavigate();

  return (
    <Tabs
      defaultActiveKey="posts"
      activeKey={props.dir}
      id="uncontrolled-tab-example"
      className="mb-3"
      fill
      onSelect={(nextTab) =>
        navigate(`/authors/${props.author_id}/${nextTab}/`)
      }
    >
      <Tab eventKey="posts" title="Posts"></Tab>

      <Tab eventKey="followers" title="Followers"></Tab>

      <Tab eventKey="friends" title="Friends"></Tab>
    </Tabs>
  );
}
