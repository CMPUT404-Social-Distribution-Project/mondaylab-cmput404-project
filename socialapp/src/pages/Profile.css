:root {
    --trickyScale: 1;
}

.profileContainer {
    display: flex;
    flex-direction: column;
    padding-left: 2rem;
    padding-right: 2rem;
    height: fit-content;
}

.profileHeader {
    background-color: var(--dark-blue);
    align-self: center;
    width: fit-content;
    min-width: 40em;
    padding-top: 2rem;
    padding-left: 2rem;
    padding-right: 2rem;
    padding-bottom: 2rem;
    border-radius: 11px;
    display: flex;
    align-items: center;
    box-shadow: rgb(0 0 0 / 16%) 1px 1px 10px;
}

.profilePicPage {
    width: 4rem;
    height: 4rem;
    border-radius: 100%;
    outline: solid var(--orange) 3px;
    text-align: center;
    display: flex;
    color: var(--white-teal);
    overflow:hidden;
    place-content: center;
}

.profilePicWithFollowButton {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 5rem;
    gap: 1rem;
}

#followButton {
    font-size: 0.8rem;
    width: fit-content;
    height: 2em;
    color: var(--darker-blue);
    background-color: var(--teal);
    display: inline-flex;
    align-items: center;
}

.profileInfo {
    display: flex;
    flex-direction: column;
    padding-left: 1rem;
    padding-right: 1rem;
    width: fit-content;
    align-content: center;
}

.profileName {
    font-size: 3rem;
    color: var(--white-teal);
    width: 100%;
    margin-left: 1rem;
    margin-bottom: 1rem;
    height: fit-content;
    line-height: 76px;
    overflow: hidden;
    text-overflow: ellipsis;
}

.profileStats {
    display: flex;
    width: 100%;
    gap: 1rem;
    margin-left: 1rem;
}

#statContainer {
    display: flex;
    gap: 0.5rem;
}

span {
    color: var(--teal);
    font-size: 1rem;
}

.infoNum {
    display: inline-flex;
    align-items: center;
    font-size: 1rem;
    text-align: center;
    align-self: center;
    padding: 0.3rem;
    border-radius: 2rem;
    width: fit-content;
    max-width: 3rem;
    height: 1.5rem;
    background-color: var(--orange);
    color: var(--dark-blue);
    overflow: hidden;
    text-overflow: ellipsis;
}

.posts {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 2em 2em 0em;
    width: 50vw;
    align-self: center;
}
