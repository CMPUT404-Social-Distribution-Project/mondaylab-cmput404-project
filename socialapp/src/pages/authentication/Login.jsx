import React, { useContext } from "react";
import "./Auth.css";
import { Formik, Form } from "formik";
import TextField from "../../components/TextField";
import * as Yup from "yup";
import Logo from "../../des/logos/logo.svg";
import LogoText from "../../des/logos/logo-text.svg";
import AuthContext from "../../context/AuthContext";
import { useNavigate } from "react-router-dom";

export default function Login() {
    const validate = Yup.object().shape({
        username: Yup.string()
            .required("Required")
            .max(26, "Invalid username"),
        password: Yup.string().required(),
    });

    const { loginUser } = useContext(AuthContext);

    const navigate = useNavigate();

    return (
        <div className="Auth-form-container">
            <div className="introduction">
                <div className="Logo-container">
                    <img src={Logo} />
                    <img id="logo-text" src={LogoText} />
                </div>
                <div id="intro-text" className="text-left">
                    Welcome to SocialDistribution,
                    <br /> your one stop shop for everything social media.
                </div>
            </div>
            <Formik
                initialValues={{
                    username: "",
                    password: "",
                }}
                validationSchema={validate}
                onSubmit={(values) => {
                    console.log(values);
                    const username = values.username;
                    const password = values.password;
                    loginUser(username, password);
                }}
            >
                {(formik) => (
                    <div>
                        <Form className="form">
                            <h3 className="Auth-form-title">Login</h3>
                            <TextField
                                label="Username"
                                name="username"
                                type="text"
                            />
                            <TextField
                                label="Password"
                                name="password"
                                type="password"
                            />
                            <div className="d-grid gap-2 mt-3">
                                <button
                                    id="submit-button"
                                    type="submit"
                                    className="btn btn-primary"
                                >
                                    Submit
                                </button>
                            </div>
                            <p
                                id="forgot"
                                className="forgot-password text-right mt-2"
                            >
                                Forgot your{" "}
                                <a id="password" href="#">
                                    password?
                                </a>
                            </p>

                            <div className="text-center">
                                New to SocialDistribution?
                            </div>
                            <div className="d-grid gap-2 mt-3">
                                <button
                                    id="signup-button"
                                    className="btn btn-primary"
                                    onClick = {() => navigate("/register")}
                                >
                                    Signup
                                </button>
                            </div>
                        </Form>
                    </div>
                )}
            </Formik>
        </div>
    );
}
