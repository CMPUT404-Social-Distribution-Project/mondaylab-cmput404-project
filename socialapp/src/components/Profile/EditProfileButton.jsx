import React, { useContext } from "react";
import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import { Formik, Form } from "formik";
import TextField from "../TextField";
import * as Yup from "yup";
import useAxios from "../../utils/useAxios.js";
import AuthContext from "../../context/AuthContext";
import "./EditProfileButton.css";

function VerticallyCenteredModal(props) {
  const validate = Yup.object().shape({
    profileImage: Yup.string().url(),
    username: Yup.string().required("Required").max(128, "Invalid username"),
    github: Yup.string().url(),
  });

  const api = useAxios();
  const { baseURL } = useContext(AuthContext);

  return (
    <Modal
      {...props}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
      contentClassName="edit-profile-modal"
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          Edit Profile
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Formik
          initialValues={{
            profileImage: `${props.author.profileImage}`,
            username: `${props.author.displayName}`,
            github: `${props.author.github}`,
          }}
          validationSchema={validate}
          onSubmit={(values) => {
            const username = values.username;
            const profileImage = values.profileImage;
            const github = values.github;

            // If fields are the same as before, don't send
            const toSend = {
              displayName: `${username}`,
              ...(profileImage === props.author.profileImage
                ? {}
                : { profileImage: `${profileImage}` }),
              ...(github === props.author.github
                ? {}
                : { github: `${github}` }),
            };

            api
              .post(`${baseURL}/authors/${props.author.uuid}/`, toSend)
              .then((response) => {
                if (response.status === 202) {
                  localStorage.setItem("loggedInUser", response.data.user);
                  props.onHide();
                  window.location.reload(); // refreshes page...not ideal.
                }
              })
              .catch((err) => {
                if (err.response) {
                  alert("Failed to submit changes. " + err.response.data);
                }
              });
          }}
        >
          {(formik) => (
            <Form>
              <div>
                <TextField
                  label="Profile Image URL"
                  name="profileImage"
                  type="text"
                />
                <TextField label="Username" name="username" type="text" />
                <TextField label="GitHub" name="github" type="text" />
                <button
                  id="submit-button"
                  type="submit"
                  className="btn btn-primary"
                >
                  Submit
                </button>
              </div>
            </Form>
          )}
        </Formik>
      </Modal.Body>
      <Modal.Footer>
        <button
          id="cancel-button"
          onClick={props.onHide}
          className="btn btn-primary"
        >
          Cancel
        </button>
      </Modal.Footer>
    </Modal>
  );
}

export default function EditProfileButton(props) {
  const [modalShow, setModalShow] = React.useState(false);
  const user_id = localStorage.getItem("user_id");

  if (user_id === props.author.uuid) {
    return (
      <div className="edit-profile">
        <Button
          className="edit-profile-button"
          variant="primary"
          onClick={() => setModalShow(true)}
        >
          Edit Profile
        </Button>

        <VerticallyCenteredModal
          show={modalShow}
          onHide={() => setModalShow(false)}
          author={props.author}
        />
      </div>
    );
  } else {
    return <></>;
  }
}
