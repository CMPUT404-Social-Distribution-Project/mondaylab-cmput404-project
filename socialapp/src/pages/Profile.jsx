import React, { useEffect, useState, useContext } from 'react';
import useAxios from "../utils/useAxios";
import "./pages.css";
import AuthContext from "../context/AuthContext";
import axios from 'axios';
import "./Profile.css";
import { useParams, useLocation } from "react-router-dom";
import PostCard from "../components/Posts/PostCard";
import UserCard from "../components/UserCard";
import EditProfileButton from "../components/Profile/EditProfileButton";
import FollowButton from "../components/Profile/FollowButton";
import ProfileTabs from "../components/Profile/ProfileTabs";
import { authorHostIsOurs, b64EncodeCredentials, createNodeObject } from '../utils/utils';
import { CgRemote } from "react-icons/cg";

function ProfilePosts(props) {
  return (
    <div className="posts-container-profile">
    {
      typeof props.postsArray.items !== 'undefined' ? 
        props.postsArray.items.map((post) => <PostCard post={post} key={post.id}/>)
        : null
    }
    </div>
  )
}

function ProfileFollowers(props) {
  return (
    <div className="followers-container-profile">
    {
      typeof props.followersArray.items !== 'undefined' ? 
        props.followersArray.items.map((follower,i) => <UserCard author={follower} key={i}/>)
        : null
    }
    </div>
  );
}

function ProfileFriends(props) {
  return (
    <div className="friends-container-profile">
    {
      typeof props.friendsArray.items !== 'undefined' ? 
        props.friendsArray.items.map((friend, i) => <UserCard author={friend} key={i}/>)
        : null
    }
    </div>
  );
}

export default function Profile() {
  const [author, setAuthor] = useState("");               // the response object we get (Author object)  
  const [postsArray, setPostsArray] = useState(""); 
  const [followersArray, setFollowersArray] = useState(""); 
  const [friendsArray, setFriendsArray] = useState(""); 
  const { baseURL } = useContext(AuthContext);      // our api url http://127.0.0.1/service
  const { author_id, dir } = useParams();                       // gets the author id in the url
  const api = useAxios();

  // node
  const [authorNode, setAuthorNode] = useState(null);     // the node object of post's author
  const [authorBaseApiURL, setAuthorBaseAPI] = useState(null);


  useEffect(() => {
    // first fetch the author
    const fetchAuthor = async () => {
      await axios
      .get(`${baseURL}/authors/${author_id}/`)
      .then((response) => {
        setAuthor(response.data);
        fetchNode(response.data);
      })
      .catch((error) => {
        console.log(error);
      });
    }
    fetchAuthor();
  }, [useLocation().state, dir])


  const fetchNode = async (author) => {
    // fetches the node object
    await api
    .get(`${baseURL}/node/?host=${author.host}`)
    .then((response) => {
      let node = createNodeObject(response, author.host);
      setAuthorNode(node);
      setAuthorBaseAPI(node.host);
    })
    .catch((err) => {
      console.log(err.response.data);
    })
  }

  // Called after rendering. Fetches data
  useEffect(() => {
    const fetchData = async (ApiURL, authorId) => {
      await api      
        .get(`${ApiURL}authors/${authorId}/posts`,
        {headers: authorNode.headers}
        )
        .then((response) => {
          setPostsArray(response.data);
        })
        .catch((error) => {
          console.log("Failed to get posts of author. " + error);
        });
      await api      
        .get(`${ApiURL}authors/${authorId}/followers`,
        {headers: authorNode.headers}
        )
        .then((response) => {
          setFollowersArray(response.data);

        })
        .catch((error) => {
          console.log("Failed to get followers of author. " + error);
        });
      await api      
        .get(`${ApiURL}authors/${authorId}/friends`,
        {headers: authorNode.headers}
        )
        .then((response) => {
          setFriendsArray(response.data);
        })
        .catch((error) => {
          console.log("Failed to get friends of author. " + error);
        });
    };
      if (!authorHostIsOurs(author.host) && authorBaseApiURL !== null) {
        fetchData(authorBaseApiURL, author_id);
      } else {
        // if the author is from our host, fetch from our API, or if something went wrong
        // trying to fetch the foreign author, then fetch that author from ours as backup.
        fetchData(baseURL+'/', author_id);
      }
      

  }, [author, authorBaseApiURL]);

  return (
    <div className="profileContainer">
      <div className="profileHeader">
        <div className="profilePicWithFollowButton">
          <div className="profilePicPage">
            <img id="profilePicPage" src={author.profileImage} alt="profilePic"/>
          </div>
          <FollowButton authorViewing={author} />
        </div>

        <div className="profileInfo">
          <div className="profileName">{author.displayName}</div>
          <div className="profileStats">
            <div id="statContainer" className="followers">
              <span>Followers:</span>
              {/* Issue with data not becoming fully available due to async operations;
              So just do 0 until we get the full info */}
              <div className="infoNum">{typeof followersArray.items === 'undefined' ? 0 : followersArray.items.length}</div>
            </div>
            <div id="statContainer" className="friends">
              <span>Friends:</span>
              <div className="infoNum">{typeof friendsArray.items === 'undefined' ? 0 : friendsArray.items.length}</div>
            </div>
            {!authorHostIsOurs(author.host) ? 
            <div className="host-indicator">
              <CgRemote style={{marginRight:"0.5em"}}/>{author.host}</div> : <div className="host-indicator" style={{background: "none"}}/>}
          </div>
          <EditProfileButton className="edit-button" author={author}/>
        </div>
      </div>
      <ProfileTabs dir={dir} author_id={author_id}/>
      {dir === 'posts' || dir === undefined ? <ProfilePosts postsArray={postsArray}/> : <></>}
      {dir === 'followers' ? <ProfileFollowers followersArray={followersArray}/> : <></>}
      {dir === 'friends' ? <ProfileFriends friendsArray={friendsArray}/> : <></>}

    </div>
  );
}
