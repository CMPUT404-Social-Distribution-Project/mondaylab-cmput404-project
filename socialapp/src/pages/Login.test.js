import React from "react";
import { fireEvent, render } from '@testing-library/react'
import Login from './authentication/Login.jsx'
import { BrowserRouter } from "react-router-dom";
import Profile from "./Profile.jsx";
import AuthProvider, { AuthContext } from "../../src/context/AuthContext"; 
describe('Testing Login:', () => {
    it("renders correctly", () => {
        const tree = render(<BrowserRouter><Login /></BrowserRouter>);
        expect(tree).toMatchSnapshot();
    });
});
// Still not work , what I have tired.
// Error before is  TypeError: Cannot destructure property 'baseURL' of '(0 , _react.useContext)(...)' as it is undefined.
// After I chnage and add Authprovider, it become to TypeError: Cannot read properties of undefined (reading '_context')
// and TypeError: render is not a function
// so, problem is when we render the page, it will first find useContext(Authprovider), then problem is happened.
// I try to mock a useContext, but they are not works.
// So I will try to make a selenium test for frontend.
// for now, noly login, register, page404 can pass the test.
/* describe('Testing Login:', () => {
    it("renders profile 1 correctly", () => {
        const tree = render(<AuthProvider><Profile /></AuthProvider>);
        expect(tree).toMatchSnapshot();
    });
});
describe('Testing Login:', () => {
    it("renders profile 2 correctly", () => {
        const tree = render(<BrowserRouter><Profile /></BrowserRouter>);
        expect(tree).toMatchSnapshot();
    });
}); */