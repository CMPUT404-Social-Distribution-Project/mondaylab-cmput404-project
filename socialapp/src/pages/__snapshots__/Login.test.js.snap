// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login: renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Auth-form-container"
      >
        <div
          class="introduction"
        >
          <div
            class="Logo-container"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
            <img
              alt="logo-text"
              id="logo-text"
              src="logo-text.svg"
            />
          </div>
          <div
            class="text-left"
            id="intro-text"
          >
            Welcome to SocialDistribution,
            <br />
             your one stop shop for everything social media.
          </div>
        </div>
        <div>
          <form
            action="#"
            class="form"
          >
            <h3
              class="Auth-form-title"
            >
              Login
            </h3>
            <div
              class="mb-2 mt-4"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="off"
                class="form-control shadow-none false"
                name="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-2 mt-4"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="off"
                class="form-control shadow-none false"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
                id="submit-button"
                type="submit"
              >
                Submit
              </button>
            </div>
            <div
              class="text-center"
              id="new-to"
            >
              New to SocialDistribution?
            </div>
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
                id="signup-button"
              >
                Signup
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Auth-form-container"
    >
      <div
        class="introduction"
      >
        <div
          class="Logo-container"
        >
          <img
            alt="logo"
            src="logo.svg"
          />
          <img
            alt="logo-text"
            id="logo-text"
            src="logo-text.svg"
          />
        </div>
        <div
          class="text-left"
          id="intro-text"
        >
          Welcome to SocialDistribution,
          <br />
           your one stop shop for everything social media.
        </div>
      </div>
      <div>
        <form
          action="#"
          class="form"
        >
          <h3
            class="Auth-form-title"
          >
            Login
          </h3>
          <div
            class="mb-2 mt-4"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="off"
              class="form-control shadow-none false"
              name="username"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-2 mt-4"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class="form-control shadow-none false"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2 mt-3"
          >
            <button
              class="btn btn-primary"
              id="submit-button"
              type="submit"
            >
              Submit
            </button>
          </div>
          <div
            class="text-center"
            id="new-to"
          >
            New to SocialDistribution?
          </div>
          <div
            class="d-grid gap-2 mt-3"
          >
            <button
              class="btn btn-primary"
              id="signup-button"
            >
              Signup
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
